src/components/PathwayCourse/index.js

```diff
@@ -76,36 +76,45 @@ const images = [
 ];
 
 function PathwayCourse() {
-  const [pathwayCourse, setPathwayCourse] = useState([]);
+  // const [pathwayCourse, setPathwayCourse] = useState([]);
   const dispatch = useDispatch();
   const user = useSelector(({ User }) => User);
-  const { data } = useSelector((state) => state.Pathways);
+  const { data } = useSelector((state) => {
+    console.log("state", state);
+    return state.Pathways;
+  });
   const classes = useStyles();
   const isActive = useMediaQuery("(max-width:" + breakpoints.values.sm + "px)");
   const params = useParams();
   const pathwayId = params.pathwayId;
 
   useEffect(() => {
-    dispatch(pathwayActions.getPathways());
-  }, [dispatch]);
-
-  useEffect(() => {
-    axios({
-      method: METHODS.GET,
-      url: `${process.env.REACT_APP_MERAKI_URL}/pathways/${pathwayId}/courses?courseType=json`,
-      headers: {
-        accept: "application/json",
-        "version-code": 40,
-        // Authorization: user.data ? user.data.token : null,
-      },
-    }).then((res) => {
-      console.log("res", res);
-      setPathwayCourse(res.data.courses);
-    });
+    // dispatch(pathwayActions.getPathways());
+    dispatch(pathwayActions.getPathwaysCourse({ pathwayId: pathwayId })); //sometimes it's returning pathways and sometimes pathway's course
```

Comment:
> //sometimes it's returning pathways and sometimes pathway's course

In response to the comment, I don't have time to investigate now, but it seems likely that this is due to a [race condition](https://en.wikipedia.org/wiki/Race_condition).  Specifically, two asynchronous API requests are being made, and both API requests will cause the same `data` property to be set.  So the response that comes back second (which could be from the first API request that's made) will be the one that determines what the data will be set to; it overwrites the data from the first response.  The reducer should be changed so that the data is written to different places in the pathways state object analogous to what's happening here where the course content data is being stored in the `data` property of `courseContent`:
https://github.com/navgurukul/bhanwari-devi/blob/main/src/components/Course/redux/reducer.js